### PR TITLE
Executor cache metrics

### PR DESCRIPTION
**Pull request by t-nelson**
#### Problem

No metrics on the executor cache

#### Summary of Changes

Add some

nvm the first four commits. I'm just staying ahead of the conflicts

### CODE REVIEW

File: runtime/src/bank.rs [hunk 1]
```diff
@@ -314,6 +314,52 @@ impl CowCachedExecutors {
...
+                ("hits", self.hits, i64),
+                ("misses", self.misses, i64),
+                ("evictions", evictions, i64),
+            );
+            debug!(
+                "Executor Cache Stats -- Hits: {}, Misses: {}, Evictions: {}",
+                self.hits, self.misses, evictions
+            );
+            if !self.evictions.is_empty() {
+                let mut evictions = self.evictions.iter().collect::<Vec<_>>();
```
Comment (Lichtso) [inline runtime/src/bank.rs:342]: I hope this won't explode in size when running on the real deal.

**Review by Lichtso — DISMISSED** [review]

File: runtime/src/bank.rs (hunk 1 above)
Comment (t-nelson) [reply runtime/src/bank.rs:342]: Yeah, same.  That's why they're trace logs instead of metrics.  Think we should cap it at something not too stupid?

Comment (sakridge) [reply runtime/src/bank.rs:342]: should we check to see if trace is enabled before doing the collecting/sorting work?

Comment (t-nelson) [reply runtime/src/bank.rs:270]: > should we check to see if trace is enabled before doing the collecting/sorting work?

Yeah good call

Comment (t-nelson) [reply runtime/src/bank.rs:271]: Done

**Pull request merged**